hr_timesheet_sheet/tests/test_hr_timesheet_reviewer_no_email.py

```diff
@@ -0,0 +1,107 @@
+            "message_type": "notification",
+        }
+
+    def test_notify_compute_recipients(self):
+        """Tests notify compute recipients
+
+        it's should prevent sending emails to the reviewer
+        """
+        timesheet = self.timesheet_model_obj.with_user(self.user_4).create(
+            {
+                "name": "Test Timesheet",
+                "employee_id": self.department_manager.id,  # user_3
+            }
+        )
+        msg_dict = self.message_dict.copy()
+        # Message created when creating timesheet
+        message_create = self.mail_message_obj.search(
+            [("model", "=", "hr_timesheet.sheet"), ("res_id", "=", timesheet.id)]
+        )
+        # send timesheet to review
+        timesheet.with_user(self.user_3).action_timesheet_confirm()
+        self.assertEqual(timesheet.state, "confirm", msg="state must be changed")
+        # find all possible reviewers
+        possible_reviewers = timesheet._get_possible_reviewers().mapped("partner_id")
+        msg_dict.update(
+            author_id=self.user_3.partner_id.id,
+            email_from=self.user_3.email,
+            res_id=timesheet.id,
+            parent_id=message_create.id,
+            subtype_id=timesheet._track_subtype(init_values=timesheet).id,
+        )
+        # Create notification message when timesheet confirmed
+        self.mail_message_obj.create(msg_dict)
+        message_confirm = self.mail_message_obj.search(
+            [
+                ("model", "=", "hr_timesheet.sheet"),
+                ("res_id", "=", timesheet.id),
+                ("subtype_id.name", "=", "Waiting Review"),
+            ]
+        )
+        # Check recipients list
+        recipients = timesheet._notify_compute_recipients(
+            message=message_confirm, msg_vals=msg_dict
+        )["partners"]
+        recipient_ids = {recipient["id"] for recipient in recipients}
+        possible_reviewer_ids = {
+            reviewer.id for reviewer in possible_reviewers if reviewer.name != "OdooBot"
+        }
+        self.assertSetEqual(
+            possible_reviewer_ids,
+            recipient_ids,
+            msg="Possible reviewers should be in recipients",
+        )
+
+        # Enable 'not send email to reviewer' flag
+        timesheet.company_id.sudo().write(
+            {
+                "hr_timesheet_reviewer_no_email": True,
+            }
+        )
+        # This user must stay in recipients
+        self.user_2.notification_type = "inbox"
+        no_email_notif_partner = self.user_2.partner_id
+
+        # Confirm timesheet
+        timesheet.with_user(self.user_4).action_timesheet_done()
+        self.assertEqual(timesheet.state, "done")
+        # Create notification message when timesheet approved
+        msg_dict.update(
+            author_id=self.user_4.partner_id.id,
+            email_from=self.user_4.email,
+            subtype_id=timesheet._track_subtype(init_values=timesheet).id,
```

Comment:
This key was set previosly

Reply:
We need to update this, because timesheet state is changed

Reply:
Ok